obs-packaging/wait-obs.sh

```diff
@@ -31,12 +33,24 @@ run_in_docker() {
 	fi
 }
 
-
 # Check all project has finshed the build
 wait_finish_building() {
-	while osc pr -q | grep '(building)'; do sleep 5; done
-	# just in case something goes wrong
-	while osc pr -q | grep '(building)'; do sleep 5; done
+	local out
+	while true; do
+		out=$(osc api "/build/${project}/_result")
+		if echo "${out}" | grep '<details>failed</details>'; then
+			echo "A build has failed"
+			osc pr
+			exit 1
+		fi
+		if echo "${out}" | grep 'state="building"'; then
+			echo "Waiting for state is not building"
```

Comment:
Suggested message:
```
echo "Project ${project} is still building, waiting"
```

You may also add a sleep to avoid API use more than necessary.